message-schemas/generated/proto/foxglove/builtins.proto

```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+package foxglove;
+
+message Time {
+  fixed32 sec = 1;
+  fixed32 nsec = 2;
+}
+
+message Duration {
+  fixed32 sec = 1;
+  fixed32 nsec = 2;
```

Comment:
I think nsec is suppose to be unsigned.

Reply:
Hm, indeed that is what ROS does. How does one handle negative durations? is -1.5sec equal to `{sec:-2,nsec:0.5e9}` or `{sec:-1,nsec:0.5e9}`?

Reply:
`{sec:-1,nsec:0.5e9}`

The sign of seconds is what matters. The nanoseconds bit is how many nanoseconds within the second.

Reply:
Are you sure?

In ROS 1, both are signed for Duration:
https://github.com/ros/roscpp_core/blob/noetic-devel/rostime/include/ros/duration.h#L75

while both are unsigned for Time:
https://github.com/ros/roscpp_core/blob/noetic-devel/rostime/include/ros/time.h#L133

See also:
https://github.com/ros/roscpp_core/blob/eabafec8f9b554d5389a1aa1b3c8145c07faaefb/rostime/src/duration.cpp#L47-L55

https://github.com/ros2/rclcpp/pull/1188


...ok, after a lot of digging, I finally found a direct example, in which they show that `-0.5sec` is represented in the .msg as `{sec:-1, nsec:500000000}`:

https://github.com/ros2/rclcpp/blob/82ddd441405a5122f1a2e445840cc2b93d713166/rclcpp/test/rclcpp/test_duration.cpp#L234-L238